apps/frontend/src/components/questionary/questionaryComponents/InstrumentPicker/QuestionaryComponentInstrumentPicker.tsx

```diff
@@ -40,14 +54,54 @@ export function QuestionaryComponentInstrumentPicker(
     question: { id, question, naturalKey },
     value,
   } = answer;
-  const [stateValue, setStateValue] = useState<Array<string> | string>(value);
+  const [stateValue, setStateValue] = useState<
+    Array<InstrumentIdAndTime> | InstrumentIdAndTime
+  >(value);
+
   const config = answer.config as InstrumentPickerConfig;
   const fieldError = getIn(errors, id);
   const isError = getIn(touched, id) && !!fieldError;
-
+  const valueWithInstrumentName = Array.isArray(value)
+    ? value.map((v: InstrumentIdAndTime) => {
+        return {
+          instrumentId: v.instrumentId,
+          instrumentName: config.instruments.find(
+            (i) => i.id.toString() === v.instrumentId
+          )?.name,
+          timeRequested: v.timeRequested,
+        };
+      })
+    : {
+        instrumentId: value?.instrumentId as string,
+        instrumentName: config.instruments.find(
+          (i) => i.id.toString() === value?.instrumentId
+        )?.name,
+        timeRequested: value?.timeRequested as string,
+      };
+  const [requestTimeForInstrument, setRequestTimeForInstrument] = useState<
+    Array<InstrumentIdNameAndTime> | InstrumentIdNameAndTime
+  >(valueWithInstrumentName);
   useEffect(() => {
     setStateValue(answer.value);
-  }, [answer]);
+    const valueWithInstrumentName = Array.isArray(value)
```

Comment:
I see this part is quite similar to the one above. Can't we reuse it somehow or put it in a function that returns the `valueWithInstrumentName`?